synapse/lib/schemas.py

```diff
@@ -199,3 +199,29 @@
     'required': ['user', 'offset', 'created', 'approved'],
     'additionalProperties': False,
 })
+
+_cellUserApiKeySchema = {
+    'type': 'object',
+    'properties': {
+        'iden': {'type': 'string', 'pattern': s_config.re_iden},
+        'name': {'type': 'string'},
+        'user': {'type': 'string', 'pattern': s_config.re_iden},
+        'created': {'type': 'integer', 'minimum': 0},
+        'updated': {'type': 'integer', 'minimum': 0},
+        'expires': {'type': ['null', 'integer'], 'minimum': 1},
+        'shadow': {
+            'type': 'object',
+        },
+    },
+    'additionalProperties': False,
+    'required': [
+        'iden',
+        'name',
+        'user',
+        'created',
+        'updated',
+        'shadow',
+        'expires',
```

Comment:
If you make `expires` optional it simplifies the schema and the kdef population logic :+1:

Reply:
updated